### PR TITLE
Feature/screen appointment schedule

### LINKED ISSUE #12: Screen | Appointment Schedule

### PR DESCRIPTION
**Pull request by jmaicaaan**
Closes #12

Hi @isotopeee , I'm not sure how to implement `Fractal` structure much. Please guide me on the changes needed. 

Btw, I've added a `BaseScreen` component to allow us have a consistent view in different screens/pages.

### CODE REVIEW

**Comment by jmaicaaan** [comment]
This pull request closes [issue](https://github.com/tinker-hub/docnest-desktop/issues/12).

File: app/screens/Appointment/AppointmentSchedule.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    margin: 1em 0;
+  }
+`;
+
+export default class AppointmentSchedule extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      appointment: {
+        schedule: new Date()
```
Comment (isotopeee) [inline app/screens/Appointment/AppointmentSchedule.js]: `new Date()` introduces a side-effect, is there a way to eliminate this?

File: app/screens/BaseScreen.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  title: PropTypes.string.isRequired
+};
+
+BaseScreen.defaultProps = {
+  actionHandler: () => {},
+  buttonIcon: <AddIcon />
+};
+
+export function BaseScreen(props) {
+  const { actionHandler, buttonIcon, children, title } = props;
```
Comment (isotopeee) [inline app/screens/BaseScreen.js]: Can we have a convention of naming the `handler props` with `on` prefix (onAction)? For event handlers, inside the components, we'll just go with the `handle` prefix just like what you did above.

File: app/screens/Appointment/AppointmentSchedule.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        title="Appointment Schedule"
+        actionHandler={this.actionHandler()}
+      >
+        <StyledSearchTextField
+          id="appointment-search-field"
+          fullWidth
+          placeholder="Search Appointment"
+        />
+        <Table
+          rowOptions={this.state.tableOptions.rowOptions}
```
Comment (isotopeee) [inline app/screens/Appointment/AppointmentSchedule.js]: Let's destructure `state` and `props` on the beginning of `render` declaration.

`render() {
  const { prop } = this.props; 
  const { state } = this.state;
// codes here
}`

File: app/components/Table/Table.js [hunk 4]
```diff
@@ -30,7 +30,11 @@ Table.defaultProps = {
     )),
   renderBodyRow: (columns, rows, rowOptions, renderBodyRowCell) =>
     rows.map(row => (
-      <MUITableRow key={row.id} {...rowOptions}>
+      <MUITableRow
+        key={row.id}
+        {...rowOptions}
+        onClick={onRowSelect(row, rowOptions.onClick)}
```
[the remark this answers is not in the document]
Comment (jmaicaaan) [reply app/components/Table/Table.js]: Updated PR! Thank you

File: app/screens/Appointment/AppointmentSchedule.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        }
+      ]
+    };
+  }
+
+  actionHandler() {
+    return () => {
+      this.setState({ open: true });
+    };
+  }
```
[the remark this answers is not in the document]
Comment (jmaicaaan) [reply app/screens/Appointment/AppointmentSchedule.js]: Correct, updated PR. 
Though I was thinking before that we might have some "extra" additional before passing to the function..

File: app/screens/Appointment/AppointmentSchedule.js (hunk 1 above)
Comment (jmaicaaan) [reply app/screens/Appointment/AppointmentSchedule.js]: I don't have any idea right now

Comment (isotopeee) [reply app/screens/Appointment/AppointmentSchedule.js]: I think this is fine for now

**Pull request merged**